File: app/src/main/java/com/jrk/mood4food/model/local/ExampleEntity.java
package com.jrk.mood4food.model.local;

import android.content.Context;

import java.util.Set;

public class ExampleEntity extends LocalEntity{

    // Attributes have to be protected
    // There are just a few possible DataTypes (See: LocalEntity.validDataTypes)

    protected String exampleString;
    protected int exampleInt;
    protected boolean exampleBoolean;
    protected long exampleLong;
    protected float exampleFloat;
    protected Set<String> exampleStringSet;

    public ExampleEntity(Context context) {
        super(context, ExampleEntity.class, true);
    }

    public String getExampleString() {
        return exampleString;
    }

    public void setExampleString(String exampleString) {
        this.exampleString = exampleString;
    }

    public int getExampleInt() {
        return exampleInt;
    }

    public void setExampleInt(int exampleInt) {
        this.exampleInt = exampleInt;
    }

    public boolean isExampleBoolean() {
        return exampleBoolean;
    }

    public void setExampleBoolean(boolean exampleBoolean) {
        this.exampleBoolean = exampleBoolean;
    }

    public long getExampleLong() {
        return exampleLong;
    }

    public void setExampleLong(long exampleLong) {
        this.exampleLong = exampleLong;
    }

    public float getExampleFloat() {
        return exampleFloat;
    }

    public void setExampleFloat(float exampleFloat) {
        this.exampleFloat = exampleFloat;
    }

    public Set<String> getExampleStringSet() {
        return exampleStringSet;
    }

    public void setExampleStringSet(Set<String> exampleStringSet) {
        this.exampleStringSet = exampleStringSet;
    }
}

File: app/src/main/java/com/jrk/mood4food/model/local/ExampleEntity.kt
package com.jrk.mood4food.model.local

import android.content.Context
import java.util.*

// Fill out the super Constructor correctly
// Please set the Param 'hasEntitySet' to true if there may be more than one Entities of this class
class ExampleEntity(context: Context) : LocalEntity(context, ExampleEntity::class.java, true) {
    // Attributes have to be protected
    // There are just a few possible DataTypes (See: LocalEntity.validDataTypes)
    var exampleString: String? = null
    var exampleInt = 0
    var isExampleBoolean = false
    var exampleLong: Long = 0
    var exampleFloat = 0f
    var exampleStringSet: Set<String>? = null

    // Attributes which can't / must not be saved in LocalStorage are marked with the Annotation '@IgnoreInStorage'
    @IgnoreInStorage
    var exampleDate: Date? = null

    @IgnoreInStorage
    var exampleIntList: List<Int>? = null

}